
## PRACTICA 1 HENRY



EMPEZAMOS IMPORTANDO LAS LIBRERIAS QUE VAMOS A USAR AUNQUE A MEDIDA QUE VALLAMOS DESARROLLANDO LA PRACTICA IREMOS AGREGANDO MAS

In [157]:
import pandas as pd

import datetime


In [ ]:
cargamos las tablas que recibimos para trabajar en el proyecto

vamos comprobando que todo se importe sin problemas y em pezamos con una limpieza elemental: 

In [158]:
precios_semana20200413=pd.read_csv("precios_semana_20200413.csv",encoding="utf-16")

In [159]:

precios_semana20200413.head()

precio    producto_id sucursal_id
0   29.90  0000000001663     2-1-014
1   29.90  0000000002288     2-1-032
2   39.90  0000000002288     2-1-096
3  499.99  0000000205870     9-1-686
4  519.99  0000000205870     9-2-248

In [160]:
precios_semana20200413.producto_id.value_counts()

7793100111563    128
7791293022505    124
7500435004664    122
7790520017420    119
7791290011205    118
                ... 
7791053528254      1
7791053518200      1
7791053504258      1
7791053433008      1
9990385651946      1
Name: producto_id, Length: 66531, dtype: int64

In [161]:
precios_semana20200413.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 472166 entries, 0 to 472165
Data columns (total 3 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   precio       472153 non-null  float64
 1   producto_id  472151 non-null  object 
 2   sucursal_id  472151 non-null  object 
dtypes: float64(1), object(2)
memory usage: 10.8+ MB


Nos cersioramos si hay duplicados y los eliminamos

In [162]:
precios_semana20200413.drop_duplicates(inplace=True)

normalizo la columna producto_id debido que deberian quedar solo con los 13 digitos ya que es un codigo EAN

In [106]:
precios_semana20200413["producto_id"]=precios_semana20200413["producto_id"].astype(str)

In [107]:
precios_semana20200413["producto_id"] = precios_semana20200413["producto_id"].apply(lambda x: x[-13:])

In [108]:
precios_semana20200518=pd.read_csv("precios_semana_20200518.txt",sep="|")

In [109]:
precios_semana20200518

precio    producto_id sucursal_id
0        29.90  0000000002288     2-1-009
1        32.90  0000000002288     2-1-037
2        36.90  0000000002288     2-1-090
3        39.90  0000000002288     2-3-247
4       499.99  0000000205870     9-1-430
...        ...            ...         ...
415288  149.99  9569753142128      25-1-1
415289   34.99  9795403001143      25-1-1
415290  312.50  9990385651922       5-1-4
415291  312.50  9990385651939       5-1-4
415292  198.90  9990385651946       5-1-4

[415293 rows x 3 columns]

In [112]:
precios_semana20200518.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 413331 entries, 0 to 415292
Data columns (total 3 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   precio       413331 non-null  float64
 1   producto_id  413331 non-null  object 
 2   sucursal_id  413331 non-null  object 
dtypes: float64(1), object(2)
memory usage: 12.6+ MB


AL HACER EL INFO EN LA TABLA ANTERIOR VEMOS QUE LA COLUMNA PRECIO TIENE CASI 2 MIL non_null, COMO VEMOS QUE EL RANGO DE PRECIOS ES BASTANTE ALTO, (no seria recomendable remplazar estos valores con la media), DECIDIMOS MEJOR ELIMINAR LOS REGISTROS DONDE HAY VALORES non_null.

In [111]:
precios_semana20200518.dropna(axis=0,inplace=True)


le damos nuevamente iinfo, y vemos que nos quedan 413331 columnas non-null

Nos cersioramos de que no hallan duplicados y los eliminamos 

In [113]:
precios_semana20200518.drop_duplicates(inplace=True)

normalizamos la columna de producto_id igual que la anterior

In [114]:
precios_semana20200413["producto_id"]=precios_semana20200413["producto_id"].astype(str)

In [115]:
precios_semana20200413["producto_id"] = precios_semana20200413["producto_id"].apply(lambda x: x[-13:])

precio    producto_id sucursal_id
0        29.90  0000000002288     2-1-009
1        32.90  0000000002288     2-1-037
2        36.90  0000000002288     2-1-090
3        39.90  0000000002288     2-3-247
4       499.99  0000000205870     9-1-430
...        ...            ...         ...
415288  149.99  9569753142128      25-1-1
415289   34.99  9795403001143      25-1-1
415290  312.50  9990385651922       5-1-4
415291  312.50  9990385651939       5-1-4
415292  198.90  9990385651946       5-1-4

[413145 rows x 3 columns]

al importar la siguiente tabla(precios_semanas_20200419_20200426.xlsx), vi que esta tenia 2 hojas de excel; (lo pude comprobar tambien cuando abri la tabla con excel), osea que era necesario extraer las tablas y luego concatenarlas ya que al parecer una es la continuacion de la otra...

esta parte no la voy a correr en vivo, ya que tarda varios minutos cargarla, separarla y concatenarla

In [116]:
precios_xlsx=pd.read_excel("precios_semanas_20200419_20200426.xlsx")

In [31]:
precios_xlsx

NameError: name 'precios_xlsx' is not defined

Acá hago la separación de las dos hojas y creo dos tablas diferentes

In [117]:
workbook = "precios_semanas_20200419_20200426.xlsx"
df_sheets = pd.ExcelFile(workbook, engine="openpyxl")

for elem in df_sheets.sheet_names:
    #print(elem)
    #print(workbook)
    df = pd.read_excel(workbook, engine="openpyxl", sheet_name=elem)
    df.to_excel(f"{elem} {workbook}.xlsx")

Aca las nombro como mencione anteriormente ya estan separadas

In [118]:
preciosxlsx=pd.read_excel("precios_20200419_20200419 precios_semanas_20200419_20200426.xlsx.xlsx")

In [119]:
preciosxlsx2=pd.read_excel("precios_20200426_20200426 precios_semanas_20200419_20200426.xlsx.xlsx")

In [45]:
preciosxlsx.head(2)

Unnamed: 0  precio sucursal_id producto_id
0           0    29.9     2-1-184        2288
1           1    39.9     2-1-206        2288

In [46]:
preciosxlsx2.head(2)

Unnamed: 0  precio sucursal_id  producto_id
0           0   399.0     2-1-092       2288.0
1           1   299.0     2-1-206       2288.0

In [ ]:
y acá las concateno, aunque con la función concat de pandas


In [164]:
precioxlsxConcat=pd.concat([preciosxlsx,preciosxlsx2],axis=0,ignore_index=True)


In [165]:
precioxlsxConcat.sucursal_id.unique()

array(['2-1-184', '2-1-206', '9-1-430', '9-2-107',
       datetime.datetime(5218, 3, 9, 0, 0), '9-1-688', '9-1-232',
       '9-2-583', '9-2-953', '9-1-645',
       datetime.datetime(5227, 3, 9, 0, 0), '10-2-286', '10-3-429',
       '15-1-1035', '9-2-161', datetime.datetime(5202, 3, 9, 0, 0),
       datetime.datetime(5206, 3, 9, 0, 0),
       datetime.datetime(2702, 1, 16, 0, 0),
       datetime.datetime(6004, 2, 16, 0, 0),
       datetime.datetime(2029, 1, 10, 0, 0), '10-2-119', '10-1-235',
       '16-1-1302', datetime.datetime(1939, 2, 9, 0, 0),
       datetime.datetime(2001, 1, 25, 0, 0), '10-1-139', '10-2-296',
       '10-3-442', '12-1-185', '12-1-215', '13-1-124', '13-1-130',
       datetime.datetime(1939, 1, 13, 0, 0),
       datetime.datetime(1962, 1, 13, 0, 0), '2-1-035', '2-1-163',
       '2-1-201', '2-2-241', '2-3-246', '2-3-247',
       datetime.datetime(2009, 1, 6, 0, 0),
       datetime.datetime(2021, 2, 6, 0, 0), '9-1-655',
       datetime.datetime(2006, 1, 10, 0, 0), '10-

Al hacer una lista de unicos en  la tabla anterior, veo que existen muchos valores que tienen formato de fecha, 
estos valores no concuerdan con la sucursal_id, asi que hago una función(def) que me ayude a normalizar estos datos, ya que puedo ver que con este formato fecha puedo deducir el verdadero sucursal_id, 

In [122]:
def idsucursal(str2):
    if type(str2)==str:
        return str2
    if type(str2)==datetime.datetime:
        str_sucursal = str2.strftime('%d/%m/%Y')
        str_nueva=str_sucursal.replace("/","-")
        lista=str_nueva
        for i in range(0,2):
            listanueva=lista.replace("0","")
    
     
        return listanueva

In [ ]:
La función la corro sobre la columna con apply y se normaliza según como yo especifique en dicha función

In [123]:
precioxlsxConcat["sucursal_id"]=precioxlsxConcat["sucursal_id"].apply(idsucursal)

In [124]:
precioxlsxConcat.sucursal_id.unique()

array(['2-1-184', '2-1-206', '9-1-430', '9-2-107', '9-3-5218', '9-1-688',
       '9-1-232', '9-2-583', '9-2-953', '9-1-645', '9-3-5227', '10-2-286',
       '10-3-429', '15-1-1035', '9-2-161', '9-3-522', '9-3-526',
       '16-1-272', '16-2-64', '1-1-229', '10-2-119', '10-1-235',
       '16-1-1302', '9-2-1939', '25-1-21', '10-1-139', '10-2-296',
       '10-3-442', '12-1-185', '12-1-215', '13-1-124', '13-1-130',
       '13-1-1939', '13-1-1962', '2-1-035', '2-1-163', '2-1-201',
       '2-2-241', '2-3-246', '2-3-247', '6-1-29', '6-2-221', '9-1-655',
       '1-1-26', '10-2-118', '47-1-1', '9-3-5216', '10-1-159', '1-1-218',
       '10-1-210', '11-5-1002', '11-5-1003', '11-5-1017', '11-5-1020',
       '11-5-1054', '11-5-363', '11-5-368', '16-1-1102', '16-1-1502',
       '18-1-25', '5-1-23', '7-1-1937', '7-1-1948', '9-1-153', '1-1-1946',
       '16-1-282', '16-1-702', '9-1-240', '9-1-243', '9-1-860',
       '9-3-5225', '9-3-5251', '10-1-142', '1-1-226', '1-1-1933',
       '1-1-1948', '1-1-1953'

Al ver que la columna Unnamed esta demás la borramos:

In [125]:
precioxlsxConcat.drop("Unnamed: 0",axis=1,inplace=True)

In [240]:
precioxlsxConcat

precio sucursal_id      producto_id
0          29.90     2-1-184             2288
1          39.90     2-1-206             2288
2         499.99     9-1-430           205870
3         539.99     9-2-107           205870
4         539.99    9-3-5218           205870
...          ...         ...              ...
937447  13999.00      25-1-1  9569753142128.0
937448   3499.00      25-1-1  9795403001143.0
937449   3125.00       5-1-3  9990385651922.0
937450   3125.00       5-1-3  9990385651939.0
937451   1989.00       5-1-3  9990385651946.0

[937452 rows x 3 columns]

En esta tabla cambiamos el orden de las columnas para que esten mas parecidas a las otras tablas.

In [126]:
precioxlsxConcat=precioxlsxConcat[["precio","producto_id","sucursal_id"]]

In [127]:

precioxlsxConcat

precio      producto_id sucursal_id
0          29.90             2288     2-1-184
1          39.90             2288     2-1-206
2         499.99           205870     9-1-430
3         539.99           205870     9-2-107
4         539.99           205870    9-3-5218
...          ...              ...         ...
937447  13999.00  9569753142128.0      25-1-1
937448   3499.00  9795403001143.0      25-1-1
937449   3125.00  9990385651922.0       5-1-3
937450   3125.00  9990385651939.0       5-1-3
937451   1989.00  9990385651946.0       5-1-3

[937452 rows x 3 columns]

array(['2-1-184', '2-1-206', '9-1-430', '9-2-107', '5218-03-09',
       '9-1-688', '9-1-232', '9-2-583', '9-2-953', '9-1-645',
       '5227-03-09', '10-2-286', '10-3-429', '15-1-1035', '9-2-161',
       '5202-03-09', '5206-03-09', '2702-01-16', '6004-02-16',
       '2029-01-10', '10-2-119', '10-1-235', '16-1-1302', '1939-02-09',
       '2001-01-25', '10-1-139', '10-2-296', '10-3-442', '12-1-185',
       '12-1-215', '13-1-124', '13-1-130', '1939-01-13', '1962-01-13',
       '2-1-035', '2-1-163', '2-1-201', '2-2-241', '2-3-246', '2-3-247',
       '2009-01-06', '2021-02-06', '9-1-655', '2006-01-10', '10-2-118',
       '47-1-1', '5216-03-09', '10-1-159', '2018-01-10', '10-1-210',
       '11-5-1002', '11-5-1003', '11-5-1017', '11-5-1020', '11-5-1054',
       '3603-05-11', '3608-05-11', '16-1-1102', '16-1-1502', '2005-01-18',
       '2003-01-05', '1937-01-07', '1948-01-07', '9-1-153', '1946-01-10',
       '2802-01-16', '16-1-702', '9-1-240', '9-1-243', '9-1-860',
       '5225-03-09', '5251-0

In [128]:
precioxlsxConcat.info() #que decidir con los nulos?

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 937452 entries, 0 to 937451
Data columns (total 3 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   precio       933909 non-null  float64
 1   producto_id  923933 non-null  object 
 2   sucursal_id  937452 non-null  object 
dtypes: float64(1), object(2)
memory usage: 21.5+ MB


al ver que cuando hacemos un info en la tabla anterior vemos que existen non_null en la columna precio, y en la columna producto_id, debemos tomar una desición al respecto de estos registros. Si decidimos hacer un dropna, estariamos perdiendo casi 17 mil registros. pero sigue siendo una cantidad muy pequeña con respecto al total de registros que hay(937452),sería lo mejor(hacer un dropna), al no encontrar valores estadisticos contundentes con qué reemplazarlos.

In [129]:
precioxlsxConcat.dropna(axis=0,inplace=True)

In [78]:
precioxlsxConcat.head(2)

precio sucursal_id producto_id
0    29.9     2-1-184        2288
1    39.9     2-1-206        2288

Hacemos el drop_duplicates para cersiorarnos de que  no hallan duplicados y ,los borramos

In [130]:
precioxlsxConcat.drop_duplicates(inplace=True)

In [128]:
precioxlsxConcat.head(3)

precio sucursal_id producto_id
0   29.90     2-1-184        2288
1   39.90     2-1-206        2288
2  499.99     9-1-430      205870

los valores de producto_id debe tener 13 digitos por que es un codigo EAN y deben cumplir esta condicion.  


debo convertir en string cada valor de producto_id para luego agregarle los ceros a la izquierda que le hagan falta 
a cada valor con una funcion lambda usando la funcion zfill()



In [131]:
precioxlsxConcat["producto_id"]=precioxlsxConcat["producto_id"].astype(str)

In [132]:
precioxlsxConcat["producto_id"]=precioxlsxConcat["producto_id"].apply(lambda x:x.zfill(13))

convierto denuevo el tipo de dato de la columna producto_id a float

In [45]:
precioxlsxConcat

precio      producto_id sucursal_id
0          29.90    0000000002288     2-1-184
1          39.90    0000000002288     2-1-206
2         499.99    0000000205870     9-1-430
3         539.99    0000000205870     9-2-107
4         539.99    0000000205870    9-3-5218
...          ...              ...         ...
937447  13999.00  9569753142128.0      25-1-1
937448   3499.00  9795403001143.0      25-1-1
937449   3125.00  9990385651922.0       5-1-3
937450   3125.00  9990385651939.0       5-1-3
937451   1989.00  9990385651946.0       5-1-3

[920488 rows x 3 columns]

vi que habian campos con el valor modificado (al comienzo del valor de 
de los 13 digitos habia otro valor,10-9-2300006000003) por esta razon aplico una función lambda para que en estos campos
solo quede el valor correspondiente(con 13 digitos)

In [133]:
precioxlsxConcat["producto_id"] = precioxlsxConcat["producto_id"].apply(lambda x: x[-13:])

In [134]:
precioxlsxConcat["producto_id"]=precioxlsxConcat["producto_id"].astype(str)

se me paso por alto esto, que algunos valores terminan en .0, entonces

quito el punto y el cero despues del punto a los datos que los tienen





In [135]:
precioxlsxConcat["producto_id"]=precioxlsxConcat["producto_id"].str.replace(".0","")

C:\Users\Usuario\AppData\Local\Temp\ipykernel_13712\1023695215.py:1: FutureWarning: The default value of regex will change from True to False in a future version.
  precioxlsxConcat["producto_id"]=precioxlsxConcat["producto_id"].str.replace(".0","")


In [136]:
precioxlsxConcat

precio  producto_id sucursal_id
0          29.90        02288     2-1-184
1          39.90        02288     2-1-206
2         499.99          058     9-1-430
3         539.99          058     9-2-107
4         539.99          058    9-3-5218
...          ...          ...         ...
937447  13999.00  69753142128      25-1-1
937448   3499.00      9501143      25-1-1
937449   3125.00    385651922       5-1-3
937450   3125.00    385651939       5-1-3
937451   1989.00    385651946       5-1-3

[920488 rows x 3 columns]

Al hacer lo anterior de nuevo pierdo el formato de producto_id, por lo tanto

paso denuevo la funcion lambda con zfill para agregar los ceros faltantes y conpletar las 13 cifras

In [137]:
precioxlsxConcat["producto_id"]=precioxlsxConcat["producto_id"].apply(lambda x:x.zfill(13))

In [138]:
precioxlsxConcat

precio    producto_id sucursal_id
0          29.90  0000000002288     2-1-184
1          39.90  0000000002288     2-1-206
2         499.99  0000000000058     9-1-430
3         539.99  0000000000058     9-2-107
4         539.99  0000000000058    9-3-5218
...          ...            ...         ...
937447  13999.00  0069753142128      25-1-1
937448   3499.00  0000009501143      25-1-1
937449   3125.00  0000385651922       5-1-3
937450   3125.00  0000385651939       5-1-3
937451   1989.00  0000385651946       5-1-3

[920488 rows x 3 columns]

precio sucursal_id    producto_id
0          29.90     2-1-184         2288.0
1          39.90     2-1-206         2288.0
2         499.99     9-1-430       205870.0
3         539.99     9-2-107       205870.0
4         539.99    9-3-5218       205870.0
...          ...         ...            ...
937447  13999.00      25-1-1  69753142128.0
937448   3499.00      25-1-1  95403001143.0
937449   3125.00       5-1-3  90385651922.0
937450   3125.00       5-1-3  90385651939.0
937451   1989.00       5-1-3  90385651946.0

[920488 rows x 3 columns]

In [227]:
precioxlsxConcat.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 920488 entries, 0 to 937451
Data columns (total 3 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   precio       920488 non-null  float64
 1   sucursal_id  920488 non-null  object 
 2   producto_id  920488 non-null  object 
dtypes: float64(1), object(2)
memory usage: 28.1+ MB


In [139]:
sucursal=pd.read_csv("sucursal.csv")

In [141]:
sucursal.head(2)

id  comercioId  banderaId      banderaDescripcion comercioRazonSocial  \
0   1-1-7           1          1              Super MAMI     Dinosaurio S.A.   
1  10-1-1          10          1  Hipermercado Carrefour            INC S.A.   

  provincia    localidad                  direccion        lat        lng  \
0      AR-X  SALSIPUEDES              E53 1011 None -31.126667 -64.295250   
1      AR-B   San Isidro  Bernardo De Irigoyen 2647 -34.491345 -58.589025   

  sucursalNombre  sucursalTipo  
0   Super Mami 4  Hipermercado  
1     San Isidro  Hipermercado

<bound method IndexOpsMixin.value_counts of 0          1-1-7
1         10-1-1
2        10-1-10
3        10-1-11
4       10-1-112
          ...   
2328    9-3-5277
2329    9-3-5626
2330    9-3-5961
2331     9-3-628
2332     9-3-662
Name: id, Length: 2333, dtype: object>

In [142]:
sucursal.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2333 entries, 0 to 2332
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   id                   2333 non-null   object 
 1   comercioId           2333 non-null   int64  
 2   banderaId            2333 non-null   int64  
 3   banderaDescripcion   2333 non-null   object 
 4   comercioRazonSocial  2333 non-null   object 
 5   provincia            2333 non-null   object 
 6   localidad            2333 non-null   object 
 7   direccion            2333 non-null   object 
 8   lat                  2333 non-null   float64
 9   lng                  2333 non-null   float64
 10  sucursalNombre       2333 non-null   object 
 11  sucursalTipo         2333 non-null   object 
dtypes: float64(2), int64(2), object(8)
memory usage: 218.8+ KB


verificamos si hay non null y los eliminamos

In [143]:
sucursal.drop_duplicates(inplace=True)

recibimos otra tabla en formato json

In [144]:
precios_semana20200503=pd.read_json("precios_semana_20200503.json")

ValueError: could not convert string to float: ''

In [146]:
precios_semana20200503

precio    producto_id sucursal_id
0         29.9  0000000002288     2-1-187
1         39.9  0000000002288     2-3-247
2       499.99  0000000205870     9-1-685
3       539.99  0000000205870      9-2-22
4       519.99  0000000205870      9-2-59
...        ...            ...         ...
397729  139.99  9569753142128      25-1-1
397730   34.99  9795403001143      25-1-1
397731   312.5  9990385651922       5-1-5
397732   312.5  9990385651939       5-1-5
397733   198.9  9990385651946       5-1-5

[397734 rows x 3 columns]

verificamos si hay duplic ados y los eliminamos

In [147]:
precios_semana20200503.drop_duplicates(inplace=True)

In [148]:
precios_semana20200503.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 397734 entries, 0 to 397733
Data columns (total 3 columns):
 #   Column       Non-Null Count   Dtype 
---  ------       --------------   ----- 
 0   precio       397734 non-null  object
 1   producto_id  397734 non-null  object
 2   sucursal_id  397734 non-null  object
dtypes: object(3)
memory usage: 12.1+ MB


igual que las anteriores normalizo la columna de producto_id

In [149]:
precios_semana20200503["producto_id"]=precios_semana20200503["producto_id"].astype(str)

In [150]:
precios_semana20200413["producto_id"] = precios_semana20200413["producto_id"].apply(lambda x: x[-13:])

ValueError: could not convert string to float: ''

In [315]:
precios_semana20200503

precio    producto_id sucursal_id
0         29.9  0000000002288     2-1-187
1         39.9  0000000002288     2-3-247
2       499.99  0000000205870     9-1-685
3       539.99  0000000205870      9-2-22
4       519.99  0000000205870      9-2-59
...        ...            ...         ...
397729  139.99  9569753142128      25-1-1
397730   34.99  9795403001143      25-1-1
397731   312.5  9990385651922       5-1-5
397732   312.5  9990385651939       5-1-5
397733   198.9  9990385651946       5-1-5

[397734 rows x 3 columns]

array([29.9, 39.9, 499.99, ..., 68.51, 142.82, 368.59], dtype=object)

recibimos otra tabla con extensión parquet
para esto debemosinstalar la librería fastparquet

In [63]:
pip install fastparquet

In [152]:
productos=pd.read_parquet("producto.parquet",engine="fastparquet")

In [153]:
productos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 72038 entries, 0 to 72037
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            72038 non-null  object
 1   marca         72036 non-null  object
 2   nombre        72036 non-null  object
 3   presentacion  72036 non-null  object
 4   categoria1    4 non-null      object
 5   categoria2    4 non-null      object
 6   categoria3    4 non-null      object
dtypes: object(7)
memory usage: 3.8+ MB


In [280]:
productos

id        marca  \
0      0000000001663   LA ANÓNIMA   
1      0000000002288   LA ANÓNIMA   
2      0000000205870    SIN MARCA   
3      0000000205894    SIN MARCA   
4      0000000205955    SIN MARCA   
...              ...          ...   
72033  9569753142128  DELI-SITAS    
72034  9795403001143         MAYO   
72035  9990385651922         DANA   
72036  9990385651939         DANA   
72037  9990385651946         DANA   

                                                  nombre presentacion  \
0                        Radicheta Atada La Anonima 1 Un       1.0 un   
1                          Perejil Atado La Anonima 1 Un       1.0 un   
2                                       Ojo de Bife 1 Kg       1.0 kg   
3                      Milanesa de Peceto Novillito 1 Kg       1.0 kg   
4                             Chiquizuela Novillito 1 Kg       1.0 kg   
...                                                  ...          ...   
72033  Milhojas Cobertura de Chocolate Blanco Deli-Si...     500.0 gr   
72034                           Mini Pizzetas Mayo 12 Un      12.0 un   
72035                 Te Negro en Hebras Lata Dana 50 Gr      50.0 gr   
72036                 Te Verde en Hebras Lata Dana 50 Gr      50.0 gr   
72037            Yerba Mate Aromatizada Lata Dana 150 Gr     150.0 gr   

      categoria1 categoria2 categoria3  
0           None       None       None  
1           None       None       None  
2           None       None       None  
3           None       None       None  
4           None       None       None  
...          ...        ...        ...  
72033       None       None       None  
72034       None       None       None  
72035       None       None       None  
72036       None       None       None  
72037       None       None       None  

[72038 rows x 7 columns]

Al hacer un info() vemos que las columnas categorias 1,2,3 casi todos son datos son non-null, tomamos la desicion
de eliminar estas columnas 

In [154]:
productos.drop(["categoria1","categoria2","categoria3"],axis=1,inplace=True)

In [155]:
productos

id        marca  \
0      0000000001663   LA ANÓNIMA   
1      0000000002288   LA ANÓNIMA   
2      0000000205870    SIN MARCA   
3      0000000205894    SIN MARCA   
4      0000000205955    SIN MARCA   
...              ...          ...   
72033  9569753142128  DELI-SITAS    
72034  9795403001143         MAYO   
72035  9990385651922         DANA   
72036  9990385651939         DANA   
72037  9990385651946         DANA   

                                                  nombre presentacion  
0                        Radicheta Atada La Anonima 1 Un       1.0 un  
1                          Perejil Atado La Anonima 1 Un       1.0 un  
2                                       Ojo de Bife 1 Kg       1.0 kg  
3                      Milanesa de Peceto Novillito 1 Kg       1.0 kg  
4                             Chiquizuela Novillito 1 Kg       1.0 kg  
...                                                  ...          ...  
72033  Milhojas Cobertura de Chocolate Blanco Deli-Si...     500.0 gr  
72034                           Mini Pizzetas Mayo 12 Un      12.0 un  
72035                 Te Negro en Hebras Lata Dana 50 Gr      50.0 gr  
72036                 Te Verde en Hebras Lata Dana 50 Gr      50.0 gr  
72037            Yerba Mate Aromatizada Lata Dana 150 Gr     150.0 gr  

[72038 rows x 4 columns]

verificamos si hay duplicados y eliminamos 

In [156]:
productos.drop_duplicates(inplace=True)

Decido concatenar todas las tablas precio para que solo quede una y dejo una para hacer una prueba para la carga incremental. 

precioxlsxConcat...... dejo esta para la prueba

In [89]:
Precio2=pd.concat([precios_semana20200413,precios_semana20200518,precios_semana20200503],axis=0,ignore_index=True)

## EXPORTAMOS CADA DATAFRAME A CSV

In [90]:
Precio2.to_csv("PrecioUnidos.csv",index=False)

In [70]:
sucursal.to_csv("Sucursal5.csv",index=False)


In [71]:
productos.to_csv("Productos6.csv",index=False)

precio    producto_id sucursal_id
0          29.9  0000000001663     2-1-014
1          29.9  0000000002288     2-1-032
2          39.9  0000000002288     2-1-096
3        499.99  0000000205870     9-1-686
4        519.99  0000000205870     9-2-248
...         ...            ...         ...
2203496  139.99  9569753142128      25-1-1
2203497   34.99  9795403001143      25-1-1
2203498   312.5  9990385651922       5-1-5
2203499   312.5  9990385651939       5-1-5
2203500   198.9  9990385651946       5-1-5

[2203501 rows x 3 columns]

para hacer la carga incremental mandamos desde aca un archivo que deje para dicha prueba, debo instalar la siguiente libreria
y con un pequeño codigo a continuacion exportamos a la base de datos el archivo correspondiente.



In [86]:
pip install PyMySQL

Note: you may need to restart the kernel to use updated packages.


In [81]:
from sqlalchemy import create_engine

In [84]:
cadena_conexion='mysql+pymysql://root:1234@localhost:3306/practica1henry'

In [87]:
conexion=create_engine(cadena_conexion)

In [91]:
precioxlsxConcat.to_sql(name='nuevatabla',con=conexion)

920488

Ya estamos listos para crear nujestra base de datos en mysql